src/browser/wisp_network.js

```diff
@@ -0,0 +1,571 @@
+"use strict";
+let wispws;
+
+let lastStream = 1;
+
+const connections = {0: {congestion: 0}};
+
+const congestedBuffer = [];
+function sendPacket(data, type, streamID) {
+    if(connections[streamID].congestion > 0) {
+        if(type === "DATA")
+            connections[streamID].congestion--;
+        wispws.send(data);
+    } else {
+        connections[streamID].congested = true;
+        congestedBuffer.push({data: data, type: type});
+    }
+}
+
+function processIncomingWispFrame(frame) {
+    // console.log(frame);
+    const view = new DataView(frame.buffer);
+    const streamID = view.getUint32(1, true);
+    switch(frame[0]) {
+        case 1: // CONNECT
+            // The server should never send this actually
+            throw new Error("Server sent client only frame: CONNECT 0x01");
+            break;
+        case 2: // DATA
+            if(connections[streamID])
+                connections[streamID].dataCallback(frame.slice(5));
+            else
+                throw new Error("Got a DATA packet but stream not registered. ID: " + streamID);
+
+
+            break;
+        case 3: // CONTINUE
+            if(connections[streamID]) {
+                connections[streamID].congestion = view.getUint32(5, true);
+            }
+
+            if(connections[streamID].congested) {
+                for(const packet of congestedBuffer) {
+                    sendPacket(packet.data, packet.type, streamID);
+                }
+                connections[streamID].congested = false;
+            }
+
+            break;
+        case 4: // CLOSE
+            if(connections[streamID])
+                connections[streamID].closeCallback(view.getUint8(5));
+            delete connections[streamID];
+            break;
+    }
```

Comment:
There should be a catch-all case with a warning message. (also in the `switch(frameObj.type) {` below)